src/entity_resolver/core/graph.py

```diff
@@ -1,16 +1,82 @@
+import re
+import collections
+from nltk.stem import PorterStemmer
+from nltk.corpus import stopwords
+
+
 class Attribute:
 
-    def __init__(self, name, attr_type, value):
+    def __init__(self, name, attr_type, value, deep_clean=True):
         self.name = name
         self.type = attr_type
-        self.value = value
+        # self.adar_amb = None
+        if self.type == 'text':
+            self._tokenize(value)
+            self.raw_value = ' '.join(self.value)
+        elif self.type == 'person_entity' and deep_clean:
+            self._clean_person_name(value)
+            self.raw_value = ' '.join(self.value)
+        else:
+            self.value = value
+            self.raw_value
+
+    def _tokenize(self, doc, to_stem=False):
+        """
+        :param doc: string of text value
+        :param to_stem: whether stemmed or not
+        :return: a lost of tokenized value
+        """
+        doc = doc.strip()
+        doc = re.sub("[^a-zA-Z]", " ", doc)
+        doc = doc.lower().split()
+        eng_stopwords = stopwords.words("english")
+        doc = [w for w in doc if w not in eng_stopwords]
+        if to_stem:
+            ps = PorterStemmer()
+            ps_stems = []
+            for word in doc:
+                ps_stems.append(ps.stem(word))
+            self.value = ps_stems
+        else:
+            self.value = doc
+
+    def _clean_person_name(self, value):
+        """
+        :param value: name string e.g. 'S.D. Whitehead' or 'Whitehead, S.D.'
+        :return: list of first name and last name parts e.g. ['whitehead', 's.d.']
+        """
+        value = value.lower()
+        cleaned_ordered_name = []  # [first name, last names]
+        if ',' in value:
+            first, last = value.split(',')
+            cleaned_ordered_name.append(first.strip())
+            last = ''.join(''.join(last.split('.')).split())
+            cleaned_ordered_name.extend(last)
+        else:
+            names = value.split()
+            first = names[-1]
+            if len(names) == 1:
+                last = ''
+            else:
+                last = ''.join(names[:-1]).strip()
+            cleaned_ordered_name.extend([first, last])
+        self.value = cleaned_ordered_name
```

Comment:
I think it is easier to not initialize `cleaned_ordered_name` before the `if else` block and simply return `[first, last]` at the end instead of using `extend` or `append`.